test/api/task.test.ts

```diff
@@ -0,0 +1,181 @@
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'POST',
+        body: '',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'POST',
+        body: '',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 500);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify({
+          name: 'Wash the cats',
+        }),
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+
+  const task = await resp!.json();
+
+  await t.step('get tasks', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'GET',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks`, {
+        method: 'GET',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+
+  await t.step('single task', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'GET',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks/${crypto.randomUUID()}`, {
+        method: 'GET',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 404);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'GET',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+
+  await t.step('update', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'PUT',
+        body: JSON.stringify({
+          isCompleted: true,
+        }),
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'PUT',
+        body: '',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 400);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'PUT',
+        body: JSON.stringify({
+          isCompleted: true,
+        }),
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+
+  await t.step('delete', async () => {
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'DELETE',
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 401);
+
+    resp = await handler(
+      new Request(`${root}/api/tasks/${task.id}`, {
+        method: 'DELETE',
+        headers,
+      }),
+      CONN_INFO,
+    );
+    assertEquals(resp.status, 200);
+  });
+});
```

Comment:
Clean up user at the end.  These records were left (output of `./clearDatabase.ts -v`:

```
[
  "refresh",
  "01HFSMCM87341P325H03TE084R",
  "01HFSMCM9BJPD9RC2GTY9887GH"
]
[ "users", "01HFSMCM87341P325H03TE084R" ]
[ "users", "name", "af2c2335-b951-4c23-8e42-bf33f4295781" ]
```

Reply:
When I run ./clearDatabase.ts -v: after testing, I still get this back: 
[
  "refresh",
  "01HG1FWFDKKNSREW318WD54VVE",
  "01HG1FWFF1FQB64J3T2J9B6HG1"
]
[ "users", "name", "notthisname" ]
I'm still researching this.